kasa/smart/modules/childdevice.py

```diff
@@ -38,11 +38,23 @@
 True
 """
 
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
 from ..smartmodule import SmartModule
 
+if TYPE_CHECKING:
+    from ..smartdevice import SmartDevice
+
 
 class ChildDevice(SmartModule):
     """Implementation for child devices."""
 
     REQUIRED_COMPONENT = "child_device"
     QUERY_GETTER_NAME = "get_child_device_list"
+
+    def __init__(self, device: SmartDevice, module: str):
+        super().__init__(device, module)
+        # Module is updated as part of device init
+        self._last_update_time = device._last_update_time
```

Comment:
Perhaps set this in the `SmartModule.__init__` to be consistent for all modules?

Reply:
Cleaned up the logic so it's all done in `_modular_update` and it's clearer what's happening.